### ISSUE TITLE
Error on collecting artifacts on tentacle after upgrading to 2.6.0.778 from 2.5.12.666

### ISSUE THREAD

**Issue by alfhenrik** [OctopusDeploy/Issues#1443]
I'm running a script through the script console that has a New-OctopusArtifact. The script execution completes without problems but the collecting artifact step fails with the following:

```
The remote host aborted the connection. This can happen when the remote server does not trust the certificate that we provided.
Pipefish.PipefishException: The remote host aborted the connection. This can happen when the remote server does not trust the certificate that we provided. ---> System.IO.IOException: Unable to write data to the transport connection: An existing connection was forcibly closed by the remote host. ---> System.Net.Sockets.SocketException: An existing connection was forcibly closed by the remote host
   at System.Net.Sockets.NetworkStream.Write(Byte[] buffer, Int32 offset, Int32 size)
   --- End of inner exception stack trace ---
   at System.Net.Sockets.NetworkStream.Write(Byte[] buffer, Int32 offset, Int32 size)
   at System.Net.Security._SslStream.StartWriting(Byte[] buffer, Int32 offset, Int32 count, AsyncProtocolRequest asyncRequest)
   at System.Net.Security._SslStream.ProcessWrite(Byte[] buffer, Int32 offset, Int32 count, AsyncProtocolRequest asyncRequest)
   at System.Net.Security.SslStream.Write(Byte[] buffer, Int32 offset, Int32 count)
   at Octopus.Shared.FileTransfer.FileSender.<Receive>b__5(Stream stream) in y:\work\refs\heads\release\source\Octopus.Shared\FileTransfer\FileSender.cs:line 110
   at Pipefish.Transport.SecureTcp.Client.SecureTcpClient.Send(SecureTcpRequest request, Action`1 response) in y:\work\3cbe05672d69a231\source\Pipefish.Transport.SecureTcp\Client\SecureTcpClient.cs:line 83
   --- End of inner exception stack trace ---
   at Pipefish.Transport.SecureTcp.Client.SecureTcpClient.Send(SecureTcpRequest request, Action`1 response) in y:\work\3cbe05672d69a231\source\Pipefish.Transport.SecureTcp\Client\SecureTcpClient.cs:line 114
   at Pipefish.Transport.SecureTcp.Streaming.Client.StreamClient.Send(Action`1 streamWriter, Int64 contentLength) in y:\work\3cbe05672d69a231\source\Pipefish.Transport.SecureTcp\Streaming\Client\StreamSender.cs:line 60
   at Octopus.Shared.FileTransfer.FileSender.Receive(SendStreamRequest message) in y:\work\refs\heads\release\source\Octopus.Shared\FileTransfer\FileSender.cs:line 122
   at Pipefish.Actor.OnReceivingTyped[TBody](Message message) in y:\work\3cbe05672d69a231\source\Pipefish\Actor.cs:line 113
Tentacle version 2.6.0.778
```

This happens on all of our polling tentacles. The listening tentacles work fine. The tentacle and server thumbprints match on all of the tentacles in the tentacle manager.

We've run this on all of our tentacles without issue before, but the error seems to have started since upgrading from 2.5.12.666 to 2.6.0.788 (we skipped some releases).

http://help.octopusdeploy.com/discussions/problems/29375

**Comment by alfhenrik** [comment]
I have replicated this in my development environment.

Install v2.5.12.666, create a listening Tentacle and a polling Tentacle, run the below script from the Script Console

```
# Copy the hosts file and give the copy a machine-specific name
$machineName = [Environment]::MachineName
$copyHostsTo = ".\$machineName-hosts.txt"
 
Copy-Item "C:\Windows\System32\drivers\etc\hosts" $copyHostsTo
 
# Tell Octopus about the artifact
New-OctopusArtifact $copyHostsTo
```

Upgrade Octopus server to 2.6.0.778, then upgrade both listening and polling Tentacle via the Octopus portal, and then run the same script as above again from the Script Console.

Result:
- Listening Tentacle works fine.
- Polling Tentacle fails with the error in the issue description.

Workaround for error:
- Reinstall the Tentacle Windows Service from the Tentacle Manager.

**Comment by DamianMac** [comment]
Closing for now.
It's an edge case and definitely fixed in 3.0 architecture
